src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -372,29 +387,160 @@ private static bool IsAnyFileSelected(IReadOnlyList<PackageItem> children)
+
+                if (libraries != null && libraries.Children != null)
+                {
+                    JSONArray jsonArray = libraries.Children.OfType<JSONArray>().First();
+
+                    string newLibrary = GetLibraryToBeInserted(libraryInstallationState);
+                    bool containsLibrary = jsonArray.BlockItemChildren.Any();
+
+                    if (containsLibrary)
+                    {
+                        insertionText = _commaDelimiter + Environment.NewLine + newLibrary + Environment.NewLine;
+                    }
+                    else
+                    {
+                        insertionText = newLibrary + Environment.NewLine;
+                    }
+
+                    int insertionIndex = libraries.AfterEnd - 1;
+
+                    if (insertionIndex > 0)
+                    {
+                        FormatSelection(textBuffer, insertionIndex, insertionText);
+                    }
+                }
+            }
+        }
+
+        private void FormatSelection(ITextBuffer textBuffer, int insertionIndex, string insertionText)
+        {
+            Controller.TextViewData viewData = Controller.TextViewConnectionListener.GetTextViewDataForBuffer(textBuffer);
+            ITextView textView = viewData.LastActiveView;
+
+            using (ITextEdit textEdit = textBuffer.CreateEdit())
+            {
+                textEdit.Insert(insertionIndex, insertionText);
+                textEdit.Apply();
+
+                IOleCommandTarget commandTarget = Shell.Package.GetGlobalService(typeof(Shell.Interop.SUIHostCommandDispatcher)) as IOleCommandTarget;
+
+                SendFocusToEditor(textView);
+
+                SnapshotSpan snapshotSpan = new SnapshotSpan(textView.TextSnapshot, insertionIndex, insertionText.Length + 1);
+                textView.Selection.Select(snapshotSpan, false);
+
+                Guid guidVSStd2K = VSConstants.VSStd2K;
+
+                commandTarget.Exec(
+                    ref guidVSStd2K,
+                    (uint)VSConstants.VSStd2KCmdID.FORMATSELECTION,
+                    (uint)OLECMDEXECOPT.OLECMDEXECOPT_DODEFAULT,
+                    IntPtr.Zero,
+                    IntPtr.Zero);
+
+                textView.Selection.Clear();
+            }
+        }
+
+        private void SendFocusToEditor(ITextView textView)
+        {
+            FrameworkElement editorWindow = (textView as IWpfTextView).VisualElement;
+            Keyboard.Focus(editorWindow);
+        }
+
+        private JSONMember GetLibraries(ITextBuffer textBuffer)
+        {
+            JSONEditorDocument document = JSONEditorDocument.FromTextBuffer(textBuffer);
+
+            if (document != null)
+            {
+                IEnumerable<JSONMember> jsonMembers = document.JSONDocument.TopLevelValue?.FindType<JSONObject>().Children?.OfType<JSONMember>();
+
+                return jsonMembers.FirstOrDefault(m => m.UnquotedNameText == ManifestConstants.Libraries);
```

Comment:
>jsonMembers [](start = 23, length = 11)

Can this be null?

Reply:
No, we wont hit this code path if it is null. If the manifest file has syntax errors we will error out much earlier in the stack.

---
In reply to: [197932421](https://github.com/aspnet/LibraryManager/pull/246#discussion_r197932421) [](ancestors = 197932421)

Reply:
Removed all ?. None of those should be null at this point actually. Updated in version 11.

---
In reply to: [197947861](https://github.com/aspnet/LibraryManager/pull/246#discussion_r197947861) [](ancestors = 197947861,197942995,197932421)